### PR TITLE
[FEAT] Remote storage FS APIs - Support for LLMWhisperer Adapters

### PR DESCRIPTION
**Pull request by harini-venkataraman**
## What
PR supports LLM Whisperer adapter for newly implemented FS APIs.
...

## Why
To support for k8s
...

## How
The fsspec instance param is passed between the methods used
...

## Relevant Docs
Not applicable
...

## Related Issues or PRs
https://github.com/Zipstack/unstract/pull/827
...

## Notes on Testing
Happy path of prompt studio.
This changes to be released with rc tag.
...

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/sdk/adapters/x2text/llm_whisperer/src/llm_whisperer.py [hunk 1]
```diff
@@ -384,7 +383,7 @@ def _write_output_to_file(
                 metadata_file_name = output_file_path.with_suffix(".json").name
                 metadata_file_path = metadata_dir / metadata_file_name
                 # Ensure the metadata directory exists
-                fs.mkdir(metadata_dir, create_parents=True)
+                fs.mkdir(str(metadata_dir), create_parents=True)
```
Comment (gaya3-zipstack) [inline src/unstract/sdk/adapters/x2text/llm_whisperer/src/llm_whisperer.py:386]: @harini-venkataraman Wondering if we have to cast metadata_dir to str or follow what we have done here in line 374 ?

**Review by gaya3-zipstack — APPROVED** [review]
Looks good. Some of these may be overlapping with @muhammad-ali-e 's  PR.

File: src/unstract/sdk/adapters/x2text/llm_whisperer/src/llm_whisperer.py (hunk 1 above)
Comment (harini-venkataraman) [reply src/unstract/sdk/adapters/x2text/llm_whisperer/src/llm_whisperer.py:386]: @gaya3-zipstack `metadata_dir` does not fetch `name` param. Do you mean this?
The way in which I have handled, passes the metadata path appended with the root path - `prompt_studio_data/<org_id>/<user_id>/<tool_id>/` and this works. I actually didnt get the optimiations you refer here.

**Review by muhammad-ali-e — APPROVED** [review]
LGTM

File: src/unstract/sdk/adapters/x2text/llm_whisperer/src/llm_whisperer.py (hunk 1 above)
Comment (gaya3-zipstack) [reply src/unstract/sdk/adapters/x2text/llm_whisperer/src/llm_whisperer.py:386]: Got it.  This is not Path lib as you mentioned in DM.

**Review by gaya3-zipstack — APPROVED** [review]
Looks good

**Pull request merged**